tez-runtime-internals/src/main/java/org/apache/tez/runtime/TezThreadDumpHelper.java

```diff
@@ -70,21 +71,17 @@ private TezThreadDumpHelper(long duration, Configuration conf) throws IOExceptio
         "path: {}", duration, basePath);
   }
 
-  public TezThreadDumpHelper() {
```

Comment:
hm, cannot recall what the purpose was of this constructor, does reflection work without this explicitly defined?
I'm afraid that as there is private parameterized constructor, class.newInstance() throws an InstantiationException, doesn't it?

Reply:
It is originally needed to instantiate [https://github.com/apache/tez/blob/ca1511998e183c8e10da79bef20463d40a39726a/tez-runtime-internals/src/main/java/org/apache/tez/runtime/TezThreadDumpHelper.java#L182-L194](https://github.com/apache/tez/blob/ca1511998e183c8e10da79bef20463d40a39726a/tez-runtime-internals/src/main/java/org/apache/tez/runtime/TezThreadDumpHelper.java#L182-L194) with zero arguments.

I think the class is not constructed in a reflective way, or it doesn't assume it's reflectively operated. I slightly updated the modifiers to make sure it
https://github.com/apache/tez/pull/324/commits/61d8249fc9051acc1b98dfe6059ea00c4634e6ce

Reply:
right, I was wrong, the hooks are created by reflection, the TezThreadDumpHelper is not
```
    helper = TezThreadDumpHelper.getInstance(conf).start(id.toString());
```